### PR TITLE
update react binding

### PR DESCRIPTION
**Pull request by ksyeo1010**
- include Porcupine directly into react
- set Wvp and peer dependency (for singleton)

due to changes in Wvp/ppn binding react doesnt do much. Any feedback is appreciated.

### CODE REVIEW

**Review by laves — APPROVED** [review]

**Review by ErisMik — APPROVED** [review]

**Review by mrrostam — APPROVED** [review]

**Review by ErisMik — APPROVED** [review]
Just make sure to check those linter warnings :) I had the same in Rhino

File: binding/react/src/use_porcupine.ts [hunk 1]
```diff
@@ -9,209 +9,133 @@
...
+  }, []);
+
+  const init = useCallback(async (
+    accessKey: string,
+    keywords: Array<PorcupineKeyword | BuiltInKeyword> | PorcupineKeyword | BuiltInKeyword,
+    model: PorcupineModel,
+    options: PorcupineOptions = {},
+  ): Promise<void> => {
+    if (options.processErrorCallback) {
+      console.warn(`'processErrorCallback' options is not supported, use 'error' state to handle errors.`);
```
Comment (laves) [inline binding/react/src/use_porcupine.ts:62]: I used a slightly different message, we should probably use the same one. I wrote:
```
"'processErrorCallback' is only supported in the Picovoice Web SDK. Use the 'error' state to monitor for errors in the React SDK."
```

File: binding/react/src/use_porcupine.ts [hunk 2]
```diff
@@ -9,209 +9,133 @@
...
-  };
+  }, [keywordDetectionCallback, errorCallback]);
 
-  const pause = (): boolean => {
-    if (webVoiceProcessor !== null) {
-      webVoiceProcessor.pause();
+  const start = useCallback(async (): Promise<void> => {
+    try {
+      if (!porcupineRef.current) {
+        setError("Porcupine not initialized");
```
Comment (laves) [inline binding/react/src/use_porcupine.ts:85]:
```suggestion
        setError("Porcupine has not been initialized or has been released");
```

File: binding/react/src/use_porcupine.ts [hunk 3]
```diff
@@ -9,209 +9,133 @@
...
-      porcupineWorkerFactory === undefined
-    ) {
-      return (): void => {
-        /* NOOP */
-      };
-    }
+  const stop = useCallback(async (): Promise<void> => {
+    try {
+      if (!porcupineRef.current) {
+        setError("Porcupine not initialized");
```
Comment (laves) [inline binding/react/src/use_porcupine.ts:101]:
```suggestion
        setError("Porcupine has not been initialized or has been released");
```

File: binding/react/src/use_porcupine.ts [hunk 4]
```diff
@@ -9,209 +9,133 @@
...
-      (Array.isArray(keywords) && keywords.length === 0)
-    ) {
-      return (): void => {
-        /* NOOP */
-      };
-    }
+  const release = useCallback(async (): Promise<void> => {
+    if (porcupineRef.current) {
+      await stop();
+      porcupineRef.current.terminate();
```
Comment (laves) [inline binding/react/src/use_porcupine.ts:117]: shouldn't we call `.release()` first? or does terminate handle that?

File: demo/react/src/VoiceWidget.tsx [hunk 5]
```diff
@@ -60,40 +48,41 @@ export default function VoiceWidget() {
...
+        Stop
       </button>
       <button
-        onClick={() => stop()}
-        disabled={isError || !isListening || !isLoaded}
+        onClick={() => release()}
+        disabled={error !== null || !isLoaded}
       >
-        Stop
+        Release
```
Comment (laves) [inline demo/react/src/VoiceWidget.tsx:85]: not sure we need a `Release` button. It's probably better to show them how to do it in a component unmount

**Review by laves — CHANGES_REQUESTED** [review]
Can you remove all the unused files from the demo? There's a bunch of analytics/test stuff in the project we don't use that just makes the project structure more confusing

File: binding/react/src/use_porcupine.ts (hunk 4 above)
Comment (ksyeo1010) [reply binding/react/src/use_porcupine.ts:117]: terminate kills the worker so no need to call `.release()`

File: binding/react/README.md [hunk 6]
```diff
@@ -1,184 +1,231 @@
...
 ## Installation
 
-Use `npm` or `yarn` to install the package and its peer dependencies. Each spoken language (e.g. 'en', 'de') is a separate package. For this example we'll use English:
+### Package
+
+Using `Yarn`:
 
 ```console
-yarn add @picovoice/porcupine-web-react @picovoice/porcupine-web-en-worker @picovoice/web-voice-processor`
+yarn add @picovoice/porcupine-web
```
Comment (laves) [inline binding/react/README.md:33]:
```suggestion
yarn add @picovoice/porcupine-react
```

File: binding/react/README.md [hunk 7]
```diff
@@ -1,184 +1,231 @@
...
 ```
 
-(or)
+or using `npm`:
 
 ```console
-npm install @picovoice/porcupine-web-react @picovoice/porcupine-web-en-worker @picovoice/web-voice-processor`
+npm install --save @picovoice/porcupine-web
```
Comment (laves) [inline binding/react/README.md:39]:
```suggestion
npm install --save @picovoice/porcupine-react
```

**Review by laves — CHANGES_REQUESTED** [review]

**Review by laves — CHANGES_REQUESTED** [review]
Don't think you addressed my above comment:
```
Can you remove all the unused files from the demo? There's a bunch of analytics/test stuff in the project we don't use that just makes the project structure more confusing
```

**Review by laves — APPROVED** [review]

**Review by mrrostam — APPROVED** [review]

**Pull request merged**